docs/resources/source_kafka.md

```diff
@@ -56,25 +56,25 @@ resource "materialize_source_kafka" "example_source_kafka" {
-- `include_partition` (Boolean) Include a partition column containing the Kafka message partition
-- `include_partition_alias` (String) Provide an alias for the partition column.
-- `include_timestamp` (Boolean) Include a timestamp column containing the Kafka message timestamp.
-- `include_timestamp_alias` (String) Provide an alias for the timestamp column.
+- `include_headers` (Boolean, Deprecated) Include message headers. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_headers_alias` (String, Deprecated) Provide an alias for the headers column. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_key` (Boolean, Deprecated) Include a column containing the Kafka message key. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_key_alias` (String, Deprecated) Provide an alias for the key column. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_offset` (Boolean, Deprecated) Include an offset column containing the Kafka message offset. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_offset_alias` (String, Deprecated) Provide an alias for the offset column. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_partition` (Boolean, Deprecated) Include a partition column containing the Kafka message partition. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_partition_alias` (String, Deprecated) Provide an alias for the partition column. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_timestamp` (Boolean, Deprecated) Include a timestamp column containing the Kafka message timestamp. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `include_timestamp_alias` (String, Deprecated) Provide an alias for the timestamp column. Deprecated: Use the new materialize_source_table_kafka resource instead.
 - `key_format` (Block List, Max: 1) Set the key format explicitly. (see [below for nested schema](#nestedblock--key_format))
 - `ownership_role` (String) The owernship role of the object.
 - `region` (String) The region to use for the resource connection. If not set, the default region is used.
 - `schema_name` (String) The identifier for the source schema in Materialize. Defaults to `public`.
-- `start_offset` (List of Number) Read partitions from the specified offset.
-- `start_timestamp` (Number) Use the specified value to set `START OFFSET` based on the Kafka timestamp.
+- `start_offset` (List of Number, Deprecated) Read partitions from the specified offset. Deprecated: Use the new materialize_source_table_kafka resource instead.
+- `start_timestamp` (Number, Deprecated) Use the specified value to set `START OFFSET` based on the Kafka timestamp. Deprecated: Use the new materialize_source_table_kafka resource instead.
```

Comment:
these two options are still currently only possible on the top-level `CREATE SOURCE` statement for kafka sources -- not yet on a per-table basis. It will require a non-trivial amount more refactoring to allow them on a per-table basis so I'm unsure if we will do that work until it's requested by a customer

Reply:
Ah yes! Good catch! Thank you!